syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.examples.iotservice";
option java_outer_classname = "IoTServiceProto";
option objc_class_prefix = "TEMP";

package chat_service;

service ChatServer {
    
    // Send Message
    rpc SendMessage (Message) returns (Confirmation){}
    // Relay Message
    rpc RelayMessage (Destination) returns (stream Message){}
}

message Message {
    string text = 1;
    string nameDestination = 2;
    string nameSender = 3;
}

message Destination {
    string ip = 1;
    string port = 2;
}

message Confirmation {
    bool confirmation = 1;
}